src/Controller/PostfinanceResponseController.php

```diff
@@ -94,33 +128,46 @@ public function processExceptionResponse(Request $request, PaymentInterface $pay
    *
    * URL of the web page to display to the customer when he cancels the payment
    * (status 1).
-   * If this field is empty the declineurl will be displayed to the customer instead.
+   * If the field is empty the declineurl will be displayed instead.
    *
    * @param \Symfony\Component\HttpFoundation\Request $request
-   *   Request
+   *   Request.
    * @param \Drupal\payment\Entity\PaymentInterface $payment
    *   The Payment Entity type.
    */
   public function processCancelResponse(Request $request, PaymentInterface $payment) {
-    $this->savePayment($payment, 'payment_cancelled');
-
-    $message = 'Postfinance communication cancelled. Payment cancelled';
-    \Drupal::logger('postfinance')->error('Processing failed with exception @e.', array('@e' => $message));
-    drupal_set_message(t('Payment processing cancelled.'), 'error');
+    try {
+      $request_data = $request->query->all();
+      if ($request_data['STATUS'] == 1) {
+        drupal_set_message(t('Payment processing cancelled.'), 'error');
```

Comment:
warning might be better here, cancel is something the user did himself, it's not an error.